app/src/main/java/com/example/footprnt/Map/Util/MapUtil.java

```diff
@@ -179,4 +197,149 @@ public static JSONObject getContinents(Activity activity){
         return mContinents;
     }
 
+    /**
+     * On search result, locates place on map
+     *
+     * @param searchText
+     * @param map
+     * @param context
+     */
+    public static void geoLocate(EditText searchText, GoogleMap map, Context context){
+        String searchString = searchText.getText().toString();
+
+        Geocoder geocoder = new Geocoder(context);
+        List<Address> list = new ArrayList<>();
+        try{
+            list = geocoder.getFromLocationName(searchString, 1);
+        } catch (IOException e){
+
+        }
+
+        if (list.size() > 0){
+            Address address = list.get(0);
+            Location l = new Location(LocationManager.GPS_PROVIDER);
+            l.setLatitude(address.getLatitude());
+            l.setLongitude(address.getLongitude());
+            centreMapOnLocation(map, l);
```

Comment:
I really like how you're starting to break out functionality into its own methods. One of the main purposes of doing this though is to isolate it from everything else so it can be reused and tested better or even replaced with something entirely different without anything needing to worry about changing their code. In this case that means we want to take a search string and return a location. We don't care if that search string came from an EditText or not and we also don't care if that location is focused on a map or not, so it would be better to have the method have an API of
  public static Location geoLocate(Context context, String searchTerm) {
and do the map centering when this returns